### PR TITLE
DEVX-2223: ccloud_library cleans up ccloud usage in ccloud_stack

### PR DESCRIPTION
**Pull request by rspurgeon**
### Description 

Removes usage of ccloud cli without JSON processing for API stability
https://confluentinc.atlassian.net/browse/DEVX-2223

### Author Validation
- [X] ccloud/ccloud-stack

### Reviewer Tasks
- [x] ccloud/ccloud-stack

### CODE REVIEW

File: utils/ccloud_library.sh [hunk 1]
```diff
@@ -1000,22 +1000,22 @@ function ccloud::destroy_ccloud_stack() {
   # Delete API keys associated to the service account
   ccloud api-key list --service-account $SERVICE_ACCOUNT_ID -o json | jq -r '.[].key' | xargs -I{} ccloud api-key delete {}
 
-  if [[ $KSQLDB_ENDPOINT != "" ]]; then
-    KSQLDB=$(ccloud ksql app list | grep $KSQLDB_NAME | awk '{print $1;}')
-    echo "KSQLDB: $KSQLDB"
-    ccloud ksql app delete $KSQLDB &>"$REDIRECT_TO"
+  if [[ "$KSQLDB_ENDPOINT" != "" ]]; then # This is just a quick check, if set there is a KSQLDB in this stack
+    local ksqldb_id=$(ccloud ksql app list -o json | jq -r 'map(select(.name == "'"$KSQLDB_NAME"'")) | .[].id')
```
Comment (ybyzek) [inline utils/ccloud_library.sh:1004]: Do we know if these `jq` functions like map and select require a specific jq version?

Comment (rspurgeon) [reply utils/ccloud_library.sh:1004]: I do not, did you encounter an issue with the changes?

Comment (rspurgeon) [reply utils/ccloud_library.sh:1004]: Changes verified with 1.6, released Nov. 2018

**Review by ybyzek — APPROVED** [review]
LGTM, nice @rspurgeon

**Pull request merged**